src/app/clusters/service-area-server/service-area-delegate.cpp

```diff
@@ -0,0 +1,101 @@
+#include "service-area-delegate.h"
+#include "service-area-server.h"
+
+using namespace chip::app::Clusters::ServiceArea;
+
+bool Delegate::GetSupportedLocationById(uint32_t aLocationId, uint32_t & listIndex, LocationStructureWrapper & aSupportedLocation)
+{
+    bool ret_value = false;
+
+    listIndex = 0;
+
+    // simple linear iteration to find the location with the desired locationID.
+    while (GetSupportedLocationByIndex(listIndex, aSupportedLocation))
+    {
+        if (aSupportedLocation.locationID == aLocationId)
+        {
+            ret_value = true;
+            break;
+        }
+
+        ++listIndex;
+    }
+
+    return ret_value;
+};
```

Comment:
is this ; needed?

Reply:
Redundant `;` were removed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/86f4b1b918ddba481eaeb5859ded257e446bed96).